pkg/defaults/defaults.go

```diff
@@ -163,6 +168,12 @@ func FromConfig(
 
 		} else if rawStep.TestStepConfiguration != nil {
 			step = steps.TestStep(*rawStep.TestStepConfiguration, config.Resources, podClient, artifactDir, jobSpec)
+		} else {
+			return nil, nil, fmt.Errorf("unhandled rawStep %s", rawStep)
```

Comment:
that will never happen

Reply:
If it will never happen, then we should use a panic to ensure that this case never happens due to programmer error.

Reply:
It's technically possible to add `[{}]` as your `raw_steps` config in CI Operator YAML, so it's a good error